src/api/posts.js

```diff
@@ -0,0 +1,17 @@
+import { request } from './api';
+
+export const getUserPosts = (posts, userId) => {
+  if (!userId) {
+    return posts;
+  }
+
+  return posts.filter(post => post.userId === userId);
+};
```

Comment:
you need to make a request for a particular user's posts, not just filter all posts fetched before
what if the new post was added by one user, but you don't see it cause you just filter the existing array on the client?